tests/resources/test_predictor.py

```diff
@@ -50,48 +54,58 @@ def test_delete():
 
 
 def test_archive_and_restore(valid_label_fractions_predictor_data):
-    session = mock.Mock()
+    session = FakeSession()
     pc = PredictorCollection(uuid.uuid4(), session)
     entity = deepcopy(valid_label_fractions_predictor_data)
     entity["metadata"]["archived"] = entity["metadata"]["created"]
-    session.put_resource.return_value = entity
+    session.set_responses(None, entity, paging_response(entity))
 
-    archived_predictor = pc.archive(uuid.uuid4())
+    with pytest.deprecated_call():
+        archived_predictor = pc.archive(uuid.uuid4())
```

Comment:
It seems like you are missing the chance to also test the new method directly.  I understand that the test cascades, but the person following after you may not grasp that.

Reply:
Good point. I'll break them out.